Tizen.native/yolo_model_training/src/yolo_model_training.c

```diff
@@ -0,0 +1,887 @@
+ */
+static void
+get_output_from_tensor_sink (const ml_tensors_data_h data,
+    const ml_tensors_info_h info, void *user_data)
+{
+  int ret = ML_ERROR_NONE;
+  size_t data_size;
+  double *output;
+  appdata_s *ad = (appdata_s *) user_data;
+
+  ret =
+      ml_tensors_data_get_tensor_data (data, 0, (void **) &output, &data_size);
+  if (ret != ML_ERROR_NONE) {
+    dlog_print (DLOG_ERROR, LOG_TAG, "ml_tensors_data_get_tensor_data failed");
+    return;
+  }
+
+  for (int i = 0; i < 4; i++) {
+    ad->result_data[i] = output[i];
+  }
+
+  dlog_print (DLOG_INFO, LOG_TAG,
+      "[training_loss: %f, training_accuracy: %f, validation_loss: %f, validation_accuracy: %f]",
+      ad->result_data[0], ad->result_data[1], ad->result_data[2],
+      ad->result_data[3]);
+
+  dlog_print (DLOG_INFO, LOG_TAG, "start >> ecore_pipe_write");
+  ecore_pipe_write (ad->ecore_pipe, ad, (unsigned int) data_size);
+  dlog_print (DLOG_INFO, LOG_TAG, "end >> ecore_pipe_write");
+}
+
+/**
+ * @brief Create an images list
+ */
+static GArray *
+create_images_list (const gchar * dir_path, appdata_s * ad)
+{
+  GDir *dir = NULL;
+  GArray *images_list = NULL;
+  const gchar *filename = NULL;
+
+  g_return_val_if_fail (dir_path != NULL, NULL);
+
+  images_list = g_array_new (FALSE, TRUE, sizeof (gchar *));
+
+  dir = g_dir_open (dir_path, 0, NULL);
+  if (!dir) {
+    dlog_print (DLOG_INFO, LOG_TAG, "Failed to open directory");
+    g_array_free (images_list, FALSE);
+    return NULL;
+  }
+
+  while ((filename = g_dir_read_name (dir)) != NULL) {
+    gchar *file_path = g_build_filename (dir_path, filename, NULL);
+    g_array_append_val (images_list, file_path);
+  }
+  g_dir_close (dir);
+
+  return images_list;
+}
+
+/**
+ * @brief Reanme images filename
+ */
+static void
+rename_images_filename (const gchar * images_path, const gchar * new_name,
+    const GArray * images_list)
+{
+  int i;
+  gchar *old_path = NULL;
+  gchar *new_path = NULL;
+
+  g_return_if_fail (images_path != NULL);
+  g_return_if_fail (new_name != NULL);
+  g_return_if_fail (images_list != NULL);
+
+  for (i = 0; i < images_list->len; i++) {
+    GString *filename = g_string_new (NULL);
+    g_string_append_printf (filename, "%s_%03d.jpg", new_name, i);
+
+    old_path = g_array_index (images_list, gchar *, i);
+    new_path = g_build_filename (images_path, filename->str, NULL);
+    if (g_rename (old_path, new_path) == 0) {
+      dlog_print (DLOG_INFO, LOG_TAG, "rename: %s -> %s", old_path, new_path);
+      //need to set label
+    }
+    g_string_free (filename, TRUE);
+  }
+}
+
+/**
+ * @brief Replace_word
+ */
+void
+replace_word (gchar * str, const char *target_word,
+    const char *replacement_word)
+{
+  g_return_if_fail (str != NULL);
+  g_return_if_fail (target_word != NULL);
+  g_return_if_fail (replacement_word != NULL);
+
+  unsigned int target_len = strlen (target_word);
+  unsigned int replacement_len = strlen (replacement_word);
```

Comment:
```suggestion
  size_t target_len = strlen (target_word);
  size_t replacement_len = strlen (replacement_word);
```

https://www.ibm.com/docs/ko/i/7.3?topic=functions-strlen-determine-string-length

Reply:
Actually, I've already seen it.:)